builder/builder.go

```
@@ -1,49 +1,148 @@
 }
+
+// applyTrustedBuildConfig overlays the contents of <testDir>/build_config onto
+// assessmentDir when present, returning a note for the build log. If the test
+// repo ships no build_config, the submission's own config is used and the note
+// records that (production test repos should always provide one).
+func applyTrustedBuildConfig(assessmentDir string, testDir string) (string, error) {
+	buildConfigDir := filepath.Join(testDir, buildConfigDirName)
+
+	info, err := os.Stat(buildConfigDir)
+	if err != nil || !info.IsDir() {
+		return fmt.Sprintf("WARNING: test repo has no %s/; building with the submission's own build config\n", buildConfigDirName), nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 104) but it returns nil

(nilerr)